### PR TITLE
GH-1266 Adds topN publish parameter

### PR DESCRIPTION
**Pull request by dtsnell4**
Adds topN publish parameter which allows user to explicitly set the number of rows displayed during pagination.

Fixes GH-1266

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

### CODE REVIEW

**Comment by dtsnell4** [comment]
@mlzummo @GordonSmith @jbrundage Please review.  This should do what you want....

http://rawgit.com/dtsnell4/Visualization/GH-1266_Table_Set_Number_Rows/demos/dermatology.html?src/other/Table.large&pagination=true&itemsPerPage=22&fixedHeader=true

**Comment by mlzummo** [comment]
looks good

**Comment by GordonSmith** [comment]
@mlzummo @dtsnell4 I think the Issue request is asking for "maxItems" not maxItems per page?  Think of it as a "TopN"?

**Comment by dtsnell4** [comment]
Sorry, I don't understand

**Comment by GordonSmith** [comment]
If you re-read:  https://github.com/hpcc-systems/Visualization/issues/1266 it has no mention of per page.

**Comment by dtsnell4** [comment]
i read "display 5 States at a time" as "display 5 States per page".  @mlzummo can you please elaborate?

**Comment by GordonSmith** [comment]
It may be ok as your fix with pagination off should work fine?

**Comment by dtsnell4** [comment]
Yes with pagination off this is never used, behaves as always

**Comment by GordonSmith** [comment]
That would be bad....  If it worked with pagination off then it would fix the issue (by my reading).

**Comment by GordonSmith** [comment]
@dtsnell4 (bump) - my understanding is that the primary use case is with pagination off and max rows set to 5?

**Comment by dtsnell4** [comment]
@GordonSmith updated 
http://rawgit.com/dtsnell4/Visualization/GH-1266_Table_Set_Number_Rows/demos/dermatology.html?src/other/Table&numRows=5

File: src/other/Table.js [hunk 1]
```diff
@@ -27,6 +27,7 @@
     Table.prototype.publishProxy("itemsPerPage", "_paginator");
     Table.prototype.publishProxy("pageNumber", "_paginator", "pageNumber",1);
     Table.prototype.publishProxy("adjacentPages", "_paginator");
+    Table.prototype.publish("numRows", null, "number", "Total number or rows of data to be displayed in the table",null,{tags:["Private"]});
```
Comment (GordonSmith) [inline src/other/Table.js]: minor:  Can you rename this to topN (as this is a common term for this feature in ECL).

**Comment by dtsnell4** [comment]
@GordonSmith done

**Comment by GordonSmith** [comment]
thx.

**Pull request merged**